source/base/mpi.cc

```diff
@@ -812,8 +814,435 @@ namespace Utilities
 #endif
     }
 
+    namespace ComputeIndexOwner
+    {
+      struct Dictionary
+      {
+        unsigned int                          dofs_per_process;
```

Comment:
Please change to `types::global_dof_index` (even though we do not expect the local size to be beyond 4B, we still use the former). Same for the other variables below that denote degrees of freedom (and not processor ranks where unsigned int is fine). For the MPI communication, you probably want to use `DEAL_II_DOF_INDEX_MPI_TYPE` as the MPI data type.

Reply:
Done in commits 41199ae9a0820d927c44cf983f9eb6f5af4cac6b, 01ff7feaaa78740803893544d63fa915d28e2c44, and 568d755442fb84598bddc04b8e3914a4a02f37d2.